kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -14,26 +14,25 @@
 
 /**
  * Abstracts the KMS operations needed for Envelope Encryption
- * @param <K> The type of Key Encryption Key id.
  * @param <E> The type of encrypted Data Encryption Key.
  */
-public interface Kms<K, E> {
+public interface Kms<E> {
```

Comment:
I'm not actually proposing that we **release** the API with  a single generic type for the edek. However I'd lean towards removing it in a separate PR.

Reply:
So I've got that, do you mean you are for waiting until the EDEK type param is removed before releasing?

Reply:
Yeah, just I think it would be odd to release the API with only one generic included. I would be happy to wait for a 0.4.0 release to be cut before merging this. I don't think its urgent.